application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/ReloadNodesService.java

```diff
@@ -0,0 +1,139 @@
+ * <p>
+ * This service periodically checks the status of nodes removes or adds nodes depending on the changes.
+ * </p>
+ */
+
+@Service
+public final class ReloadNodesService implements DisposableBean
+{
+
+    private static final Logger LOG = LoggerFactory.getLogger(ReloadNodesService.class);
+    private static final int DEFAULT_SCHEDULER_AWAIT_TERMINATION_IN_SECONDS = 60;
+    private final EccNodesSync myEccNodesSync;
+    private final DistributedJmxConnectionProvider myJmxConnectionProvider;
+    private final DistributedNativeConnectionProvider myDistributedNativeConnectionProvider;
+    private final ScheduledExecutorService myScheduler = Executors.newScheduledThreadPool(1);
+    private final Interval reLoadInterval;
+    private final NodeListComparator nodeListComparator = new NodeListComparator();
+
+    public ReloadNodesService(final EccNodesSync eccNodesSync,
+                                 final Config config,
+                                 final DistributedJmxConnectionProvider jmxConnectionProvider,
+                                 final DistributedNativeConnectionProvider distributedNativeConnectionProvider
+    )
+    {
+        this.myEccNodesSync = eccNodesSync;
+        this.myJmxConnectionProvider = jmxConnectionProvider;
+        this.myDistributedNativeConnectionProvider = distributedNativeConnectionProvider;
+        this.reLoadInterval = config.getConnectionConfig().getReloadPolicy();
+    }
+
+    @PostConstruct
+    public void startScheduler()
+    {
+        long reLoadIntervalInMills = reLoadInterval.getInterval(TimeUnit.MILLISECONDS);
+        LOG.info("Starting ReloadNodesService with reLoadInterval={} ms", reLoadIntervalInMills);
+        myScheduler.scheduleWithFixedDelay(this::reloadNodes, reLoadIntervalInMills, reLoadIntervalInMills, TimeUnit.MILLISECONDS);
+    }
+
+    @VisibleForTesting
+    private void reloadNodes()
+    {
+        List<Node> oldNodes = myDistributedNativeConnectionProvider.getNodes();
+        List<Node> newNodes = myDistributedNativeConnectionProvider.reloadNodes();
+        List<NodeChangeRecord> nodeChangeList = nodeListComparator.compareNodeLists(oldNodes, newNodes);
+        if (!nodeChangeList.isEmpty())
+        {
+            myDistributedNativeConnectionProvider.setNodes(newNodes);
+            Iterator<NodeChangeRecord> iterator = nodeChangeList.iterator();
+            while (iterator.hasNext())
+            {
+                NodeChangeRecord nodeChangeRecord = iterator.next();
+                if (nodeChangeRecord.getType() == NodeChangeRecord.NodeChangeType.INSERT)
+                {
+                    processInsertRecord(nodeChangeRecord);
+                }
+                if (nodeChangeRecord.getType() == NodeChangeRecord.NodeChangeType.DELETE)
+                {
+                    processDeleteRecord(nodeChangeRecord);
+                }
+            }
```

Comment:
There is nothing here yet of when an ip address changes.  Should I treat it as a delete of the old ip address and an insert for the new ip address